src/utils/date.ts

```diff
@@ -21,6 +32,42 @@ export const isBeforeDate = (
   return dateToCheck < dateToCheckAgainst;
 };
 
+export const conversationsListDate = (dateToTransform: Date): string => {
```

Comment:
I think this whole file might be more complicated than necessary. All we need to do is check if the date is today, yesterday, or within 7 days and then return a formatted string. We don't need to do the formatting manually, JavaScript provides this [functionality for us](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString).

Reply:
I did a refactor to make it a bit more compact. Let me know if any more refactoring should be done here

Reply:
Working with dates is pretty tricky, so it's ideally best to use native tooling instead of writing your own versions of things.